setup.py

```diff
@@ -136,12 +136,36 @@ def run_tests(self):
     'tox',
 ]
 
+"""This section defines feature flags end-users can use in dependencies"""
 extras_require = {
-    """This section defines feature flags end-users can use in dependencies"""
-    # Enables memcached session support via `cherrypy[memcached-session]`:
-    'memcached-session': ['python-memcached>=1.58'],
+    'doc': [
+        'docutils',
+        'sphinx_rtd_theme',
+    ],
+    'json': ['simplejson'],
+    'routes_dispatcher': ['routes>=2.3.1'],
+    'ssl': ['pyOpenSSL'],
+    'test_tools': [
+        'coverage',  # inspects tests coverage
+        # TODO: drop nose dependency in favor of py.test analogue
+        'nose',  # only used in cherrypy.test.{helper,test_{compat,routes}}
+        'nose-testconfig',  # only used in cherrypy.test.helper
+        'objgraph',  # cherrypy.lib.gctools
+    ],
+    # Enables memcached session support via `cherrypy[memcached_session]`:
+    'memcached_session': ['python-memcached>=1.58'],
+    'multienv_tests': tests_require,
+    'xcgi': ['flup'],
+
+    # http://docs.cherrypy.org/en/latest/advanced.html?highlight=windows#windows-console-events
+    ':sys_platform == "win32"': ['pywin32'],
 }
 
+if sys.version_info < (3, 3):
```

Comment:
FYI, we would be able to avoid this conditional declaration using `backports.unittest_mock` if only we were using pytest as the runner (work that's underway in another branch).